python-spec/src/somacore/query/query.py

```diff
@@ -587,11 +588,16 @@ def _var_df(self) -> data.DataFrame:
 
     @property
     def _threadpool(self) -> futures.ThreadPoolExecutor:
-        """Creates a thread pool just in time."""
+        """
+        Returns the threadpool provided by the experiment's context.
+        If not available, creates a thread pool just in time."""
+        if (
+            self.experiment.context is not None
+            and self.experiment.context.threadpool is not None
```

Comment:
nit: since `context` and `context.threadpool` will be nonzero if they are set at all, this can be

```python
context = self.experiment.context
if context and context.threadpool:
  return context.threadpool
```

without the need for `is not None`.